src/controllers/message/services/MessageServiceFactory.ts

```diff
@@ -0,0 +1,22 @@
+import { ConnectionRecord } from '@credo-ts/core'
+import { InjectQueue } from '@nestjs/bull'
+import { Injectable, Optional } from '@nestjs/common'
+import { Queue } from 'bull'
+
+import { IBaseMessage } from '../../../model'
+
+import { CoreMessageService } from './CoreMessageService'
+
+@Injectable()
+export class MessageServiceFactory {
+  constructor(
+    @Optional() @InjectQueue('message') private messageQueue: Queue,
+    private readonly coreMessageService: CoreMessageService,
+  ) {}
+
+  async setProcessMessage(redisAvailable: boolean, message: IBaseMessage, connection: ConnectionRecord) {
+    return redisAvailable
+      ? await this.messageQueue.add('', { message, connection })
```

Comment:
What does that '' mean? Seems a bit awkward to me.

Reply:
its a [conditional ternary operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_operator). It's used to simplify the "if" condition

Reply:
I'm asking about `await this.messageQueue.add('', { message, connection })`. The first argument is an empty string. What does it mean? Is it the right way of doing it?

Reply:
You're right, it's not necessary